OOP/OOP6_ResizableArray/src/main/java/LukCharList.java

```diff
@@ -0,0 +1,150 @@
+public class LukCharList implements CharList {
+
+    private char[] array;
+    private char[] temporaryArray;
+
+    public LukCharList(){
+        array = new char[0];
+        temporaryArray = new char[0];
+    }
+
+    public int save(char charToSave) {
+        if(this.array.length > 0) {
+            temporaryArray = this.array;
+        }
+        this.array = new char[this.array.length + 1];
+        rewrite(temporaryArray);
+        array[this.array.length - 1] = charToSave;
+        return this.array.length - 1;
+    }
+
+    private void rewrite(char[] temporary) {
```

Comment:
IMHO this would be cleaner if you would return result of this method, instead of modifying a class field (which shouldn't be there anyway;) )

Reply:
@MatDls modified